solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -496,17 +496,29 @@ contract RandomBeacon is Ownable {
 
     /// @notice Approves DKG result. Can be called after challenge period for the
     ///         submitted result is finished. Considers the submitted result as
-    ///         valid, pays reward to the result submitter and completes the
-    ///         group creation by activating the candidate group.
+    ///         valid, pays reward to the result submitter, punishes misbehaved
+    ///         group members and completes the group creation by activating the
+    ///         candidate group.
     function approveDkgResult() external {
         dkg.approveResult();
-        // Pay the DKG result submission reward.
         tToken.safeTransfer(dkg.resultSubmitter, dkgResultSubmissionReward);
+
+        uint32[] memory misbehavedMembers = groups.getLatestMisbehavedMembers();
+        if (misbehavedMembers.length > 0) {
+            //TODO: Replace 2 weeks with governable parameter
+            punishOperators(misbehavedMembers, 2 weeks);
+        }
+
         groups.activateCandidateGroup();
         dkg.complete();
 
-        // TODO: Handle DQ/IA. Should result submitter receive
-        //       reward if they failed to call this function?
+        // TODO: Should result submitter receive reward if they failed to call
+        //       this function?
```

Comment:
@pdyraga How do we go about this one?
I don't think we should handle this TODO in this PR, but make a task and handle it in another PR.

Reply:
The cost of `approveDkgResult` is:
* `142930` if there are no misbehaved members
* `186578` when there are 5

It will go higher when we implement the sortition-pool's banFromRewards function